### ISSUE TITLE
Certain pronoun classes need inflectional paradigms

### ISSUE THREAD

**Issue by aarppe** [UAlbertaALTLab/cree-intelligent-dictionary#164]
It would make sense to create "paradigm" tables for personal pronouns, demonstrative pronouns, as well as interrogative pronouns - these "paradigms" could be invoked from any of their constituent member pronouns, presenting how that pronoun is positioned in relation to the rest in the same subclass. This is related to #127. Don't know about the need to contrast between basic and full paradigms, though.

**Comment by aarppe** [comment]
There are now tentative layouts for personal and demonstrative pronouns, to be found in giella/langs/crk as:

```
inc/paradigms/non-linguistic/pronoun-demonstrative-basic.layout.tsv
inc/paradigms/non-linguistic/pronoun-personal-basic.layout.tsv
```

What needs to be checked is whether one strictly needs the dynamic FST-based generation of the pronominal forms, as in this case they can be and have been easily enumerated, but will the presentation of static forms work with the current interpretation of `*.layout` files?

**Comment by aarppe** [comment]
@eddieantonio You wanted these a while back. As said, for these I'm not generating the forms at all, since the forms per cells are static. Of course, we could still generate them, if that would be useful or necessary for some purpose. There's a file for that in:

`inc/paradigms/pronoun.paradigm`

**Comment by eddieantonio** [comment]
Thanks for this! I'm not super sure what the most elegant way would be to incorporate this into itwêwina would be at the moment, but this is definitely useful!

**Comment by eddieantonio** [comment]
A note about this:

The current layout paradigm parser considers anything in quotes to be a label or title.

So this:

```
        "One"
"I/me/my"       "niya"
"you (one)/your"        "kiya"
"s/he/it/him/her/his/her/its"   "wiya"
        "Many"
"we (but not you)/us/our"       "niyanân"
"you and we/us/our"     "kiyânaw"
"you (all)/your"        "kiyawâw"
"they/them/their"       "wiyawâw"
```

Is a table full of labels, and none of them are actual "forms".

**Comment by aarppe** [comment]
I was toying with various options, first from the old itwêwina standpoint, and in a two column layout I had more abstract row/column labels for `Speaker/Listener/Other` and `One/Many`, and then one could have used the treatment of the cells as labels to provide the exact translation of the cell content (and to distinguish niya with correponding row/column labels of `Speaker + One` as 'I' and `Speaker + Many` as `we`.

However, for the current implementation we use only a single column, and then the row label can provide the exact translation. But even in that case, one doesn't really need to generate the forms, and FST allows for many personal pronouns two variants, e.g. _niya_ and _nîya_ for _I_, which I was trying to avoid. Of course, we may opt to show such variation.

If one unquotes the actual forms in the cells, i.e. having `niya` instead of `"niya"`, will the current implementation be able to interpret and present them properly? (I'm not advocating for changing the code, much or at all).

**Comment by aarppe** [comment]
Concerning linking the two paradigms with itwêwina, for the demonstratives the animate vs. inanimate use the same lemma, i.e. _awa_ vs. _ôma_. However, for the personal pronouns the lemma is the same as the form (or one of the two variants).

I don't know exactly how paradigm generation is invoked coding-wise for nouns and verbs currently, and how that might be applied, or bended, for pronouns, given the above FST coding. I suppose the `+Pers` and `+Dem` (and eventually `+Interr`) features could be used to invoke the corresponding paradigm tables.

Also, do note that there is only one paradigm layout for the personal vs. demonstrative pronouns for now. Maybe for personal pronouns one might put the inclusive ones such as nîsta 'me too' into the extended one. And maybe for demonstratives, one could select only a few as the basic ones. But in this first implementation I opted to just show all without any division, just for us to have something to work with, though this can be changed later on.

**Comment by aarppe** [comment]
The layouts for personal and demonstrative pronouns, according to the new scheme, can now be found in ohci-itwêwina:

```
CreeDictionary/res/layouts/pronoun-demonstrative-basic.layout.tsv
CreeDictionary/res/layouts/pronoun-personal-basic.layout.tsv
```

These follow two strategies: 1) Personal pronouns are not generated but merely spelled out (not in quotes, in contrast to labels); 2) Demonstrative pronouns are generated based on their linguistic feature specifications (similar to nouns and verbs). One can easily swap to using either strategy for both layouts, depending on what is best in the coding implementation.

What I'm not entirely sure about is how we would link any personal or demonstrative pronoun to their layouts, i.e. whether the current scheme for nouns and verbs would be applicable to pronouns as well.

**Comment by eddieantonio** [comment]
Fixed as of #768/#769
